File: src/main/java/afm/utils/OnClose.java
package afm.utils;

import lombok.Getter;

import afm.database.Database;
import afm.user.Settings;

// Singleton
public class OnClose extends Thread {

	@Getter(lazy = true)
	private static final OnClose instance = new OnClose();

	private OnClose() {
		setName("On close thread");
		setDaemon(false);
	}

	// Save Settings preferences & Save MyList & ToWatch into database
	@Override
	public void run() {
		Settings.save();
		Database.saveAll();
	}

}

File: src/main/java/afm/utils/OnClose.kt
package afm.utils

import afm.database.Database
import afm.user.Settings

object OnClose : Thread() {

    init {
        name = "On close thread"
        isDaemon = true
    }

    // Save Settings preferences & Save MyList & ToWatch into database
    override fun run() {
        Settings.save()
        Database.saveAll()
    }

}
